modules/core/src/system.cpp

```diff
@@ -1835,7 +1835,15 @@ void* TLSDataContainer::getData() const
     {
         // Create new data instance and save it to TLS storage
         pData = createDataInstance();
-        getTlsStorage().setData(key_, pData);
+        try
+        {
+            getTlsStorage().setData(key_, pData);
+        }
+        catch (...)
+        {
+            delete pData;
```

Comment:
`delete` can't be used here.
Use `deleteDataInstance(pData)` instead.

Reply:
Thanks for the reply. Yes, this is a report from static analyzer.